client/battle/BattleFieldController.cpp

```diff
@@ -212,8 +212,8 @@ std::set<BattleHex> BattleFieldController::getHighlightedHexesStackRange()
 		result.insert(hex);
 
 	// display the movement shadow of the stack at b (i.e. stack under mouse)
-	const CStack * const shere = owner.curInt->cb->battleGetStackByPos(hoveredHex, false);
-	if(shere && shere != owner.stacksController->getActiveStack() && shere->alive())
+	const CStack * const shere = owner.curInt->cb->battleGetStackByPos(hoveredHex, true);
+	if(shere && shere != owner.stacksController->getActiveStack())

```

Comment:
what does `shere` mean? "stack here"?

Reply:
Yes, "stack on currently hovered hex".
I usually replace such well-selected variable names, but missed this one somehow
(and that parameter that got changed from false to true is "select only alive anits")

Reply:
we should start from variables such as CSH, CCH, LOCPLINT, VLC.... xD

Reply:
What's wrong with those?
CSH - CServerHandler
CCS - CClientState (not CCH by the way)
LOCPLINT - LOCal PLayer INTerface
VLC - VCMI Library Classes 

COMPLETELY obvious!
Seriously though, largest problem with them is not name, but rather their singleton status. Especially for VLC. Nothing good comes from global state. 
We could have easily implement (for example) single app mode without namespace hacks or "view enemy turn" option. But global state makes those much harder than it should be